src/pkg/docs/model/query.go

```
@@ -9,7 +11,13 @@ type QueryRequest struct {
 }
 
 func (q *QueryRequest) Validate() bool {
-	return len(q.Query) != 0
+	if len(q.Query) == 0 {
+		return false
+	}
+
+	_, err := aqlprocessor.NewAqlProcessor(q.Query).Process()
+
+	return err == nil
```

Let's return the error here and output its text to the log in the handler